scripts/aggregate-themes/theme-aggregator.js

```diff
@@ -156,6 +91,84 @@ class ThemeAggregator {
     return assets;
   }
 
+  /**
+   * Triggers aggregation and generation for a given theme.
+   * @param {string} themeName - The theme to aggregate.
+   * @param {Object} options - The aggregation options.
+   * @param {boolean} defaultFlag - Whether the theme to be generated is a root or scope theme. Guards against default theme and scope theme being equivalent.
+   * @returns {array} - An array of file names.
+   */
+  static triggerAggregationAndGeneration(themeName, options = {}, defaultFlag) {
+    if (!themeName) return null;
+    Logger.log(`Aggregating ${themeName} files...`, { LOG_CONTEXT });
+    const { theme, scoped = [] } = options;
+    const isRoot = themeName === theme && defaultFlag;
+    const isScoped = scoped.indexOf(themeName) > -1;
```

Comment:
Won't this always be true unless this is the root theme we are aggregating?

Reply:
Flattened this to a isRoot flag. Use this to flex between root and scoped themes.

1d1bb30#diff-1672fd55492cdce050f5aa314b17029bR72
1d1bb30#diff-1672fd55492cdce050f5aa314b17029bR108